requests/src/main/scala/weco/api/requests/responses/LookupPendingRequests.scala

```diff
@@ -20,31 +16,38 @@ trait LookupPendingRequests extends CustomDirectives {
   implicit val ec: ExecutionContext
 
   def lookupRequests(patronNumber: SierraPatronNumber): Route = {
-    val userHolds =
-      for {
-        userHolds <- sierraService.getStacksUserHolds(patronNumber)
-
-        holdsWithCatalogueIds <- Future.sequence(
-          userHolds.right.get.holds.map { hold: StacksHold =>
-            itemLookup
-              .bySourceIdentifier(hold.sourceIdentifier)
-              .map {
-                case Left(elasticError: ElasticsearchError) =>
-                  warn(
-                    s"Unable to look up $hold in Elasticsearch: ${elasticError}"
-                  )
-                  hold
-
-                case Right(item: Item[IdState.Identified]) =>
-                  hold.copy(item = Some(item))
-              }
-          }
+    val itemHolds = for {
+      holdsMap <- sierraService.getHolds(patronNumber)
+
+      itemLookupResults <- Future
+        .sequence(
+          holdsMap.keys
+            .map(itemLookup.bySourceIdentifier)
+            .map(_.map {
+              case Right(item) => Some(item)
+              case Left(err) =>
+                warn(s"$err")
```

Comment:
We should put some context on these warnings, so we can more easily trace them to a line of code.

Do we get the source identifier we were trying to look up?